core-rust/state-manager/src/state_manager.rs

```diff
@@ -586,119 +590,59 @@ where
                 Consider wiping your database dir and trying again.", top_txn_identifiers.0);
         }
 
-        let genesis_data_chunks_len = genesis_data_chunks.len();
-        let num_genesis_txns = genesis_data_chunks_len + 2; // + bootstrap + wrap up
+        let mut named_system_transactions = Vec::new();
+        named_system_transactions.push((
```

Comment:
So we previously prepared then executed / stored one transaction at a time, which kept the memory footprint low.

Preparing them all in advance is _probably_ fine (given that Lukasz crunched the numbers of mainnet), but this does increase the memory pressure quite a lot...

It looks like we could change it back to execute one transaction at a time? If you feel this is better, we should probably get Lukasz in to double-check re the memory overhead for mainnet migration.

Reply:
I agree, let's not put memory pressure where not needed.

I now tried to solve it using a classic, simple, effective and battle-proven "lazy provider" approach (which would take like ~10s in java).
In rust, I got completely lost somewhere between `Vec<(&'static str, Box<dyn FnOnce() -> ...>)>` and `reference "chunk" may not live long enough` 🤷‍♂️ .

So instead, I unrolled the loop, and it is 3 lines less than when rolled 🤯  

I believe this completely addresses the concern, so I am going to merge after CI passes.

Reply:
Haha yes, I love a good loop unroll in situations like this. Rust often punishes us for doing silly abstractions, which I've come to view as a feature, not a bug :)

Reply:
> silly abstractions

![image](https://github.com/radixdlt/babylon-node/assets/119413677/4d012a87-458b-4734-99ba-e327c49cfa1b)

You have insulted an entire Java community, and especially **The `LazyProviderFactoryManager` Club** ®️ 

_(conceptually, a loop which goes over a polymorphic collection of things [== different genesis transactions] which are all supposed to be treated the same way [== prepared and committed] is a very valid thing to me, and not necessarily worth unrolling; but yeah, I see cons of this approach in rust)_

Reply:
Haha! I meant silly for rust of course 😇